src/main/java/com/example/ninthhomework/controller/UpdateForm.java

```diff
@@ -0,0 +1,24 @@
+package com.example.ninthhomework.controller;
+
+import com.example.ninthhomework.domain.user.model.Characters;
+
+public class UpdateForm {
+
+    private String name;
+
+    private Integer age;
+
+    public Characters UpdateForm(int id) {
+        Characters updatecharacter = new Characters(id, name, age);
+        return updatecharacter;
+    }
```

Comment:
Charactersへの変更処理でしょうか？
```suggestion
    public Characters toCharacters(int id) {
        Characters updatecharacter = new Characters(id, name, age);
        return updatecharacter;
    }
```
のようにするといいです。

ただし、ControllerとServiceの更新処理を見ると
```
ControllerでUpdateFormをCharacterに変換してServiceに渡す。
Serviceでは引数のCharacterを使わないでMapperから取得したCharacterを使って更新する。
引数はなぜかCharacterなのにupdateFormという変数名になっていて、それを更新後の値として利用する。
```

Controller、Serviceでキャラクターのオブジェクトをどう扱うかですが、実装方法はいろいろあります。

たとえば、

ControllerからUpdateFormをServiceに渡すときに引数を1つずつ渡す。

```java
public interface CharactersService {
  public Characters updateCharacter(int id, String name, Integer age);
略
}
```
Controllerの処理を次のようにします。
```java
@PatchMapping("/characters/{id}")
public ResponseEntity<Map<String, String>> update
        (@PathVariable("id") int id, @RequestBody UpdateForm updateForm) {
    charactersService.updateCharacter(id, updateForm.getName, updateForm.getAge());
    return ResponseEntity.ok(Map.of("message", "character successfully updated"));
}
```

Reply:
294da3ddcf6983478f1c32268709f7647ee95230
修正しました。